apps/modernization-api/src/test/java/gov/cdc/nbs/investigation/InvestigationResolverTest.java

```diff
@@ -0,0 +1,105 @@
+
+    @Captor
+    private ArgumentCaptor<InvestigationFilter> filterCaptor;
+
+    @Captor
+    private ArgumentCaptor<Long> patientIdCaptor;
+
+    @Captor
+    private ArgumentCaptor<Pageable> pageableCaptor;
+
+    private final int maxPageSize = 23;
+
+    private InvestigationResolver resolver;
+
+    @BeforeEach
+    void setup() {
+        MockitoAnnotations.openMocks(this);
+        resolver = new InvestigationResolver(maxPageSize, finder);
+    }
+
+    @Test
+    void should_pass_filter() {
+        var filter = InvestigationTestUtil.createFilter();
+        resolver.findInvestigationsByFilter(filter, new GraphQLPage(maxPageSize));
+        verify(finder).find(filterCaptor.capture(), Mockito.any());
+        var actualFilter = filterCaptor.getValue();
+        assertEquals(filter.getPatientId(), actualFilter.getPatientId());
+        assertEquals(filter.getConditions(), actualFilter.getConditions());
+        assertEquals(filter.getProgramAreas(), actualFilter.getProgramAreas());
+        assertEquals(filter.getJurisdictions(), actualFilter.getJurisdictions());
+        assertEquals(filter.getPregnancyStatus(), actualFilter.getPregnancyStatus());
+        assertEquals(filter.getEventIdType(), actualFilter.getEventIdType());
+        assertEquals(filter.getEventId(), actualFilter.getEventId());
+        assertEquals(filter.getEventDateSearch(), actualFilter.getEventDateSearch());
+        assertEquals(filter.getCreatedBy(), actualFilter.getCreatedBy());
+        assertEquals(filter.getLastUpdatedBy(), actualFilter.getLastUpdatedBy());
+        assertEquals(filter.getProviderFacilitySearch(), actualFilter.getProviderFacilitySearch());
+        assertEquals(filter.getInvestigatorId(), actualFilter.getInvestigatorId());
+        assertEquals(filter.getInvestigationStatus(), actualFilter.getInvestigationStatus());
+        assertEquals(filter.getOutbreakNames(), actualFilter.getOutbreakNames());
+        assertEquals(filter.getCaseStatuses(), actualFilter.getCaseStatuses());
+        assertEquals(filter.getNotificationStatuses(), actualFilter.getNotificationStatuses());
+        assertEquals(filter.getProcessingStatuses(), actualFilter.getProcessingStatuses());
+    }
+
+    @Test
+    void should_pass_patientId() {
+        Long patientId = 123L;
+        resolver.findOpenInvestigationsForPatient(patientId, new GraphQLPage(maxPageSize));
+        verify(finder).findOpenInvestigations(patientIdCaptor.capture(), Mockito.any());
+        var actualPatient = patientIdCaptor.getValue();
+        assertEquals(patientId, actualPatient);
+    }
+
+    @Test
+    void should_limit_max_page_size() {
+        Long patientId = 123L;
+        var requestedPageSize = maxPageSize + 1;
+        QueryException ex = null;
+        try {
+            resolver.findOpenInvestigationsForPatient(patientId, new GraphQLPage(requestedPageSize));
+        } catch (QueryException e) {
+            ex = e;
+        }
+        assertNotNull(ex);
```

Comment:
Exceptions can be tested with `assertThatThrownBy` from assertj or the existence of the exception should fail the test.

```java
assertThatThrownBy(() -> resolver.findOpenInvestigationsForPatient(patientId, new GraphQLPage(requestedPageSize)))
.isInstanceOf(QueryException.class);
```
Other assertions can be chained if you wanting to test the message also.  I'd also argue that this is a test of a feature of the `GraphQLPage` object :upside_down_face: